pkg/platform/runtime/model/buildplanner.go

```diff
@@ -0,0 +1,299 @@
+
+func (bp *BuildPlanner) FetchBuildResult(commitID strfmt.UUID, owner, project string) (*BuildResult, error) {
+	resp := &model.BuildPlan{}
+	err := bp.client.Run(request.BuildPlan(owner, project, commitID.String()), resp)
+	if err != nil {
+		return nil, errs.Wrap(err, "failed to fetch build plan")
+	}
+
+	// Check for errors in the response
+	if resp.Project.Type == model.NotFound {
+		return nil, locale.NewError("err_buildplanner_project_not_found", "Build plan does not contain project")
+	}
+	if resp.Project.Commit.Type == model.NotFound {
+		return nil, locale.NewError("err_buildplanner_commit_not_found", "Build plan does not contain commit")
+	}
+	if resp.Project.Commit.Build.Type == model.BuildResultPlanningError {
+		var errs []string
+		var isTransient bool
+		for _, se := range resp.Project.Commit.Build.SubErrors {
+			errs = append(errs, se.Message)
+			isTransient = se.IsTransient
+		}
+		return nil, &BuildPlannerError{
+			wrapped:          locale.NewError("err_buildplanner", resp.Project.Commit.Build.Error),
+			validationErrors: errs,
+			isTransient:      isTransient,
+		}
+	}
+
+	if resp.Project.Commit.Build.Status == model.Planning {
+		resp, err = bp.pollBuildPlan(owner, project, commitID.String())
+		if err != nil {
+			return nil, errs.Wrap(err, "failed to poll build plan")
+		}
+	}
+
+	// The type aliasing in the query populates the
+	// response with emtpy targets that we should remove
+	removeEmptyTargets(resp)
+
+	// Extract the available platforms from the build plan
+	var bpPlatforms []strfmt.UUID
+	for _, t := range resp.Project.Commit.Build.Terminals {
+		if t.Tag == model.TagOrphan {
+			continue
+		}
+		bpPlatforms = append(bpPlatforms, strfmt.UUID(strings.TrimPrefix(t.Tag, "platform:")))
+	}
+
+	// Get the platform ID for the current platform
+	platformID, err := platformModel.FilterCurrentPlatform(HostPlatform, bpPlatforms)
+	if err != nil {
+		return nil, locale.WrapError(err, "err_filter_current_platform")
+	}
+
+	// Filter the build terminals to only include the current platform
+	var filteredTerminals []*model.NamedTarget
+	for _, t := range resp.Project.Commit.Build.Terminals {
+		if platformID.String() == strings.TrimPrefix(t.Tag, "platform:") {
+			filteredTerminals = append(filteredTerminals, t)
+		}
+	}
+	resp.Project.Commit.Build.Terminals = filteredTerminals
+
+	buildEngine := Alternative
+	for _, s := range resp.Project.Commit.Build.Sources {
+		if s.Namespace == "builder" && s.Name == "camel" {
+			buildEngine = Camel
+			break
+		}
+	}
+
+	res := BuildResult{
+		BuildEngine: buildEngine,
+		Build:       resp.Project.Commit.Build,
+		BuildReady:  resp.Project.Commit.Build.Status == model.Ready,
+		CommitID:    strfmt.UUID(resp.Project.Commit.CommitID),
+	}
+
+	// If the build is alternative the buildLogID type will identify it as a recipe ID.
+	// The other buildLogID type is for camel builds which we don't use for builds in progress.
+	// There should one be one build log ID for alternative builds.
```

Comment:
Please clean up this sentence.

Reply:
Should be better now. Let me know if that's more clear.